src/compilation/mod.rs

```diff
@@ -0,0 +1,79 @@
+use crate::util::read_file_contents;
+
+use self::{
+    swaypad::{clean_error_content, create_project, remove_project, write_main_file},
+    tooling::{build_project, check_forc_version, switch_fuel_toolchain},
+};
+use hex::encode;
+use std::fs::read_to_string;
+
+pub mod swaypad;
+pub mod tooling;
+
+/// Build and destroy a project.
+pub fn build_and_destroy_project(
+    contract: String,
+    toolchain: String,
+) -> (String, String, String, String) {
+    // Check if any contract has been submitted.
+    if contract.is_empty() {
+        return (
+            "".to_string(),
+            "".to_string(),
+            "No contract.".to_string(),
+            "".to_string(),
+        );
+    }
+
+    // Switch to the given fuel toolchain and check forc version.
+    switch_fuel_toolchain(toolchain);
+    let forc_version = check_forc_version();
+
+    // Create a new project.
+    let project_name = create_project().unwrap();
```

Comment:
should we change this to `expect` so if it panics, a message is printed to the console

Reply:
There is a lot of room for improvement regarding error handling in the server. Opened https://github.com/FuelLabs/sway-playground/issues/27